sdk/attestation/azure-security-attestation/test/samples/attestation/attestation.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief This sample provides the code implementation to use the Key Vault Certificates SDK client
+ * for C++ to create, get, update, delete and purge a certificate.
+ *
+ * @remark The following environment variables must be set before running the sample.
+ * - ATTESTATION_AAD_URL:  Points to an Attestation Service Instance in AAD mode.
+ * - ATTESTATION_ISOLATED_URL:  Points to an Attestation Service Instance in Isolated mode.
+ * - LOCATION_SHORT_NAME:  Specifies the short name of an Azure region to use for shared mode
+ * operations.
+ * - AZURE_TENANT_ID:     Tenant ID for the Azure account. Used for authenticated calls to the
+ * attestation service.
+ * - AZURE_CLIENT_ID:     The Client ID to authenticate the request. Used for authenticated calls to
+ * the attestation service.
+ * - AZURE_CLIENT_ID:     The Client ID to authenticate the request. Used for authenticated calls to
+ * the attestation service.
+ * - AZURE_CLIENT_SECRET: The client secret. Used for authenticated calls to the attestation
+ * service.
+ *
+ */
+
+#include "get_env.hpp"
+
+#include "attestation_collateral.hpp"
+#include <azure/attestation.hpp>
+#include <azure/core/base64.hpp>
+#include <azure/identity.hpp>
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+using namespace Azure::Security::Attestation;
+using namespace Azure::Security::Attestation::Models;
+using namespace std::chrono_literals;
+// cspell: words MRENCLAVE MRSIGNER
+namespace Azure { namespace Security { namespace Attestation { namespace Samples {
+
+  /** @brief Simple call to attestation specifying a predefined SGX quote.
+   *
+   * Note that calls to AttestSgxEnclave are anonymous and do not require authentication.
+   */
+  void SampleAttestSgxEnclaveSimple()
```

Comment:
move fn implementations to the end, so people can easily see main() and later navigate to any of the functions

Reply:
there's no convention.  I am just looking for an easy and quick finding the program starting point (without scrolling). But, up to you Sir.

Reply:
We're almost certainly going to end with one sample executable per scenario, so that means we're going to just have "main".